src/ranker/ShortestDistanceRanker.cpp

```diff
@@ -11,52 +11,34 @@ namespace feeding {
 
 //==============================================================================
 void ShortestDistanceRanker::sort(
-    std::vector<std::unique_ptr<FoodItem>>& items) const
-{
+    std::vector<std::unique_ptr<FoodItem>> &items) const {
   // Ascending since score is the distance.
   TargetFoodRanker::sort(items, SORT_ORDER::ASCENDING);
 }
 
 //==============================================================================
 std::unique_ptr<FoodItem> ShortestDistanceRanker::createFoodItem(
-    const aikido::perception::DetectedObject& item,
-    const Eigen::Isometry3d& forqueTransform) const
-{
+    const aikido::perception::DetectedObject &item,
+    const Eigen::Isometry3d &forqueTransform) const {
   TiltStyle tiltStyle(TiltStyle::NONE);
   double rotation = 0.0;
 
-  // Get Ideal Action Per Food Item
-  auto it = std::find(FOOD_NAMES.begin(), FOOD_NAMES.end(), item.getName());
-  if (it != FOOD_NAMES.end())
-  {
-    int actionNum = BEST_ACTIONS[std::distance(FOOD_NAMES.begin(), it)];
-    switch (actionNum / 2)
-    {
-      case 1:
-        tiltStyle = TiltStyle::VERTICAL;
-        break;
-      case 2:
-        tiltStyle = TiltStyle::ANGLED;
-        break;
-      default:
-        tiltStyle = TiltStyle::NONE;
-    }
-    rotation = (actionNum % 2 == 0) ? 0.0 : M_PI / 2.0;
+  // Get Action From DetectedObject
+  std::string itemAction = item.getInfoByKey<std::string>("action");
```

Comment:
@egordon If I'm understanding this correctly, the code now uses the action and rotation that is outputted by food detection. Is there no longer any way to override it, particularly the tiltStyle, using the YAML? (Because if you recall, even the code earlier had hardcoded best actions -- it seems like the new code has no way to override the action or tilt style outputted by perception).

Reply:
See below comment, the override now happens on a higher level rather than in the Ranker.

This also means that now all Rankers (ShortestDistance and SuccessRate) function identically, pulling the action from perception to be overridden (optionally) later.